kb/0003-scripting-functions.md

```diff
@@ -5,7 +5,711 @@ _[back to Index](../index.md)_
+    syntax:
+        bool CheckAttribute(object &obj, string attribute);
+    ```
+
+* **argb**
+    * **Compiler Token**: `FUNC_ARGB`
+    ``` C++
+    syntax:
+        int argb(int a, int r, int g, int b);
+    ```
+
+* **DeleteEntities**
+    * **Compiler Token**: `FUNC_DELETE_ENTITIES`
+    ``` C++
+    syntax:
+        void DeleteEntities();
+    ```
+
+* **ClearEvents**
+    * **Compiler Token**: `FUNC_CLEAR_EVENTS`
+    ``` C++
+    syntax:
+        void ClearEvents(); 
+    ```
+
+* **SaveEngineState**
+    * **Compiler Token**: `FUNC_SAVEENGINESTATE`
+    ``` C++
+    syntax:
+        void SaveEngineState(1);
+    ```
+
+* **LoadEngineState**
+    * **Compiler Token**: `FUNC_LOADENGINESTATE`
+    ``` C++
+    syntax:
+        void LoadEngineState(); /1/?
+    ```
+
+* **Event**
+    * **Compiler Token**: `FUNC_EVENT`
+    ``` C++
+    syntax:
+        void Event(); 
+    ```
+
+* **PostEvent**
+    * **Compiler Token**: `FUNC_POSTEVENT`
+    ``` C++
+    syntax:
+        void PostEvent(); 
+    ```
+
+* **fts**
+    * **Compiler Token**: `FUNC_FTS`
+    ``` C++
+    syntax:
+        string fts(float, int); 
+    ```
+
+* **ClearPostEvents**
+    * **Compiler Token**: `FUNC_CLEAR_POST_EVENTS`
+    ``` C++
+    syntax:
+        void ClearPostEvents();
+    ```
+
+* **SetArraySize**
+    * **Compiler Token**: `FUNC_SET_ARRAY_SIZE`
+    ``` C++
+    syntax:
+        void SetArraySize(2); 
+    ```
+
+* **GetAttributeValue**
+    * **Compiler Token**: `FUNC_GET_ATTRIBUTE_VALUE`
+    ``` C++
+    syntax:
+        string GetAttributeValue(aref attribute);
+    ```
+
+* **Vartype**
+    * **Compiler Token**: `FUNC_VARTYPE`
+    ``` C++
+    syntax:
+        string Vartype(aref attribute);
+
+    /* TODO(yakvi): actually confirm it's an aref input, from the source looks like it 
+       Unfortunately, I'm not yet well versed in reading this code. Isn't 
+       `pV->IsReference()` can only refer to an aref or any value at all?
+
+        ================
+        pV = SStack.Pop();
+        if (!pV) { ... }
+        pV = pV->GetReference();
+        if (!pV)
+        {
+            SetError(INVALID_FA); // "Invalid function argument"
+            break;
+        }
+        if (pV->IsReference())
+            strcpy_s(sVarName, "ref:");
+        else
+            sVarName[0] = 0;
+        pV = pV->GetVarPointer();
+        switch (pV->GetType())
+        {
+            // print out return type
+        }
+        pV = SStack.Push();
+        pV->Set(sVarName);
+        pVResult = pV;
+        return pV; // return to user
+        ================
+
+        Will need to go in and compare with others that I know are arefs.
+    */
+    ```
+
+* **Breakpoint**
+    * **Compiler Token**: `FUNC_BREAKPOINT`
+    ``` C++
+    syntax:
+        void Breakpoint();
+    ```
+
+* **Pow**
+    **Compiler Token**: `FUNC_POW`
+    ``` C++
+    syntax:
+        float Pow(float exponent, float base);
+        // TODO(yakvi): base/exponent inverted??? to verify
```

Comment:
@mitrokosta, could you help here as well?

Reply:
from looking at the [code](https://github.com/storm-devs/storm-engine/blob/develop/src/apps/ENGINE/internal_functions.cpp#L1256) and comparing with the signature of [pow](https://www.cplusplus.com/reference/cmath/pow/) it seems to be the case indeed.
@mitrokosta what do you think?

Reply:
I'll leave this TODO there for now. It's still a stub, and I'll be able to verify it by the time I get to the function.

Reply:
Sorry for taking so long to answer. It is not inverted. I verified it in scripts: Log_Info(""+pow(10.0, 3.0)); yields 1000. Because of the nature of stack data structure all function arguments are inverted in engine.

Reply:
That's incredibly useful info that I will use for the multiparam functions! I was suspecting something like this, and having a confirmation is absolutely perfect.